Source/SIMPLib/DataArrays/StructArray.hpp

```diff
@@ -113,6 +113,33 @@ class StructArray : public IDataArray
     return ptr;
   }
 
+  /**
+   * @brief Static constructor
+   * @param numElements The number of elements in the internal array.
+   * @param name The name of the array
+   * @return Std::Shared_Ptr wrapping an instance of StructArrayTemplate<T>
+   */
+  static Pointer CreateArray(size_t numElements, const std::string& name, bool allocate = true)
+  {
+    if(name.empty())
+    {
+      return NullPointer();
+    }
+    StructArray<T>* d = new StructArray<T>(numElements, true);
+    if(allocate)
+    {
+      if(d->Allocate() < 0)
+      {
+        // Could not allocate enough memory, reset the pointer to null and return
+        delete d;
+        return StructArray<T>::NullPointer();
+      }
+    }
+    d->setName(QString::fromStdString(name));
+    Pointer ptr(d);
```

Comment:
If you store 'd' as a shared pointer initially, you don't have to manually delete it when returning NullPointer()